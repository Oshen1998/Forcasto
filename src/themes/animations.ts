export const animations = {
  loading: require('../assets/animations/loading.json'),
  warning: require('../assets/animations/warning.json'),
  cloudLoading: require('../assets/animations/cloudLoading.json'),
  underDev: require('../assets/animations/underDev.json'),
};
