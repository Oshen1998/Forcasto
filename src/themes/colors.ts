export const LightColors = {
  Background: {
    AVATAR_EDIT: '#006FFD',
    PRIMARY: '#FFFFFF',
    LIGHT_THEME: '#80E2C9',
    BLACK_TRANSPARENT: 'rgba(0, 0, 0, 0.5)',
    NOTIFICATION: 'rgba(92,208,130,0.1)',
    LINE: '#C5C6CC',
    BIOMETRICS: '#F0F0F0',
    THEME: '#5CD082',
    ICON: '#898A8D',
    WARNING: 'rgba(255, 14, 0, 0.1)',
    SWIPE_CARD: '#EFFFF5',
    SWIPE_CARD_DELETE: '#FF665D',
    LIGHT_DOCUMENT: '#F1FFF6',
    BLACK_BUTTON: '#1D1B20',
  },
  Border: {
    DEFAULT: '#FFFFFF',
    PRIMARY: '#F0F0F0',
    TEXT_INPUT: '#BEBEBE',
    TEXT_INPUT_DEFAULT_BORDER: '#E2E8F0',
    TEXT_INPUT_FOCUSED_BORDER: '#94A3B8',
    THEME: '#5CD082',
  },
  Icon: {
    PRIMARY: '#000000',
    ACCENT: '#FFFFFF',
    THEME: '#5CD082',
    ERROR: '#FF5252',
    DISABLED: '#71727A',
    BLENDING: '#5CD082',
  },
  Text: {
    ACCENT: '#FFFFFF',
    DESCRIPTION: '#64748B',
    DISABLED: '#71727A',
    ERROR: '#FF5252',
    ICON_BUTTON: '#324A5E',
    LIGHT_DISABLED: '#F0F0F0',
    LIGHT_THEME: '#CAFFDB',
    PRIMARY: '#000000',
    THEME: '#5CD082',
  },

  TRANSPARENT: 'transparent',
};
