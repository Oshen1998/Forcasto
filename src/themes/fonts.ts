export const FONT_SIZES = {
  SuperTitle: 64,
  HeroTitle: 40,
  BigTitle: 32,
  Title: 28,
  Subtitle: 24,
  MediumTitle: 22,
  SmallTitle: 18,
  Caption: 16,
  Body: 14,
  Tiny: 12,
  BitTiny: 11,
  Footnote: 10,
};

export const FONT_FAMILIES = {
  Manrope: {
    ExtraLight: 'Manrope-ExtraLight',
    Light: 'Manrope-Light',
    Regular: 'Manrope-Regular',
    Medium: 'Manrope-Medium',
    SemiBold: 'Manrope-SemiBold',
    Bold: 'Manrope-Bold',
    ExtraBold: 'Manrope-ExtraBold',
  },
};
