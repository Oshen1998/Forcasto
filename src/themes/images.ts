import {ImageSourcePropType} from 'react-native';

export const images = {
  backgrounds: {
    //
  },
  weather: {
    lighteningWithRain:
      require('../assets/images/rainWIthThunder/img.png') as ImageSourcePropType,
  },
  icons: {
    location:
      require('../assets/icons/location/icon.png') as ImageSourcePropType,
    chevron: require('../assets/icons/chevron/icon.png') as ImageSourcePropType,
    reload: require('../assets/icons/reload/icon.png') as ImageSourcePropType,
    sunny: require('../assets/icons/sun/icon.png') as ImageSourcePropType,
    moon: require('../assets/icons/moon/icon.png') as ImageSourcePropType,
    rainyNight:
      require('../assets/icons/rain_night/icon.png') as ImageSourcePropType,
    rainyDay:
      require('../assets/icons/rain_day/Rainy.png') as ImageSourcePropType,
    snowNight:
      require('../assets/icons/snow_night/icon.png') as ImageSourcePropType,
    snowDay:
      require('../assets/icons/snow_day/Snowy.png') as ImageSourcePropType,
    humidity:
      require('../assets/icons/humidity/icon.png') as ImageSourcePropType,
    wind: require('../assets/icons/wind/icon.png') as ImageSourcePropType,
  },
};
