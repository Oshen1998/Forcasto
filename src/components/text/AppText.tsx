import {ColorValue, StyleProp, Text, TextStyle} from 'react-native';
import React from 'react';
import {FONT_FAMILIES} from '../../themes/fonts';

type AppTextProps = {
  fontWeight?:
    | 'normal'
    | 'bold'
    | '100'
    | '200'
    | '300'
    | '400'
    | '500'
    | '600'
    | '700'
    | '800'
    | '900'
    | 100
    | 200
    | 300
    | 400
    | 500
    | 600
    | 700
    | 800
    | 900
    | 'ultralight'
    | 'thin'
    | 'light'
    | 'medium';
  fontFamily?: string;
  fontSize?: number;
  fontColor?: ColorValue;
  textAlign: 'center' | 'left' | 'right';
  children: any;
  numberOfLines?: number;
  ellipsizeMode?: 'head' | 'middle' | 'tail' | 'clip';
  style?: StyleProp<TextStyle>;
};

const AppText = ({
  fontColor = '#000000',
  fontSize = 12,
  fontWeight = '300',
  fontFamily = FONT_FAMILIES.Manrope.Medium,
  children = '',
  textAlign = 'left',
  numberOfLines = 1,
  ellipsizeMode = 'tail',
  style,
}: AppTextProps) => {
  return (
    <Text
      numberOfLines={numberOfLines}
      ellipsizeMode={ellipsizeMode}
      style={
        style || {
          color: fontColor,
          fontFamily: fontFamily,
          fontSize: fontSize,
          fontWeight: fontWeight,
          textAlign: textAlign,
        }
      }>
      {children}
    </Text>
  );
};

export default AppText;
