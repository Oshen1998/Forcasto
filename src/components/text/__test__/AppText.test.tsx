import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {FONT_FAMILIES} from '../../../themes/fonts';
import AppText from '../AppText';

describe('AppText Component', () => {
  it('renders correctly', () => {
    render(
      <AppText textAlign="center" testId="default-text">
        Default Text
      </AppText>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should applies custom styles correctly', () => {
    const customStyle = {
      color: 'red',
      fontSize: 16,
      fontWeight: 'bold',
      textAlign: 'center',
    };

    const {getByTestId} = render(
      <AppText
        testId="custom-text"
        fontColor="red"
        fontSize={16}
        fontWeight="bold"
        textAlign="center">
        Custom Text
      </AppText>,
    );

    const textElement = getByTestId('custom-text');
    expect(textElement.props.style).toMatchObject(customStyle);
  });

  it('should renders multiple lines correctly', () => {
    const {getByTestId} = render(
      <AppText textAlign="center" testId="multiline-text" numberOfLines={3}>
        This is a long text that should span multiple lines
      </AppText>,
    );

    const textElement = getByTestId('multiline-text');
    expect(textElement.props.numberOfLines).toBe(3);
  });

  it('should applies custom font family correctly', () => {
    const {getByTestId} = render(
      <AppText
        textAlign="center"
        testId="custom-font"
        fontFamily={FONT_FAMILIES.Manrope.Bold}>
        Custom Font Text
      </AppText>,
    );

    const textElement = getByTestId('custom-font');
    expect(textElement.props.style.fontFamily).toBe(FONT_FAMILIES.Manrope.Bold);
  });

  it('should applies custom ellipsizeMode correctly', () => {
    const {getByTestId} = render(
      <AppText
        textAlign="center"
        testId="ellipsize-text"
        ellipsizeMode="middle">
        This text will be ellipsized in the middle if it's too long
      </AppText>,
    );

    const textElement = getByTestId('ellipsize-text');
    expect(textElement.props.ellipsizeMode).toBe('middle');
  });

  it('should applies custom style prop correctly', () => {
    const customStyle = {
      letterSpacing: 2,
      color: '#000000',
    };

    const {getByTestId} = render(
      <AppText
        textAlign="center"
        testId="custom-style-text"
        style={customStyle}>
        Custom Styled Text
      </AppText>,
    );

    const textElement = getByTestId('custom-style-text');
    expect(textElement.props.style).toMatchObject(customStyle);
  });
});
