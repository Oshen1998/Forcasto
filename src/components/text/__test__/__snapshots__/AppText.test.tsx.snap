// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText Component renders correctly 1`] = `
<Text
  accessibilityLabel="default-text"
  ellipsizeMode="tail"
  numberOfLines={1}
  style={
    {
      "color": "#000000",
      "fontFamily": "Manrope-Medium",
      "fontSize": 12,
      "fontWeight": "300",
      "textAlign": "center",
    }
  }
  testID="default-text"
>
  Default Text
</Text>
`;
